cunumeric/thunk.py

```diff
@@ -216,6 +216,13 @@ def diag(self, rhs, extract, k, stacklevel):
         """
         raise NotImplementedError("Implement in derived classes")
 
```

Comment:
diag_helper is not mentioned in the base class

Reply:
Actually, it looks like `thunk/deferred/eager.py:diagonal` is no longer used. You should remove the corresponding code, and rename `thunk:diag_helper` to `thunk:diagonal`.